lib/mapper/product_mapper.rb

```diff
@@ -2,134 +2,36 @@ module BigcommerceProductAgent
-                    is_default: product['isDefault'],
-                    type: product['isDigital'] == true || is_digital ? 'digital' : 'physical',
-                    description: product['description'] || '',
-                    price: product['offers'] && product['offers'][0] ? product['offers'][0]['price'] : '0',
-                    categories: self.get_categories(product),
-                    availability: self.get_availability(product),
-                    weight: product['weight'] ? product['weight']['value'] : '0',
-                    width: product['width'] ? product['width']['value'] : '0',
-                    depth: product['depth'] ? product['depth']['value'] : '0',
-                    height: product['height'] ? product['height']['value'] : '0',
-                    meta_keywords: self.meta_keywords(product),
-                    meta_description: self.meta_description(product) || '',
-                    search_keywords: self.get_search_keywords(additional_data.delete(:additional_search_terms), product),
-                    is_visible: variant ? false : true,
-                    preorder_release_date: product['releaseDate'] && product['releaseDate'].to_datetime ? product['releaseDate'].to_datetime.strftime("%FT%T%:z") : nil,
-                    preorder_message: self.get_availability(product) == 'preorder' ? product['availability'] : '',
-                    is_preorder_only: self.get_availability(product) == 'preorder' ? true : false,
-                    page_title: product['page_title'] || '',
+                  availability: self.get_availability(product),
+                  categories: self.get_categories(product),
+                  depth: product['depth'] ? product['depth']['value'] : '0',
+                  description: product['description'] || '',
+                  height: product['height'] ? product['height']['value'] : '0',
+                  is_default: product['isDefault'],
+                  is_preorder_only: self.get_availability(product) == 'preorder' ? true : false,
+                  is_visible: true,
+                  meta_description: self.meta_description(product) || '',
+                  meta_keywords: self.meta_keywords(product),
+                  name: name,
+                  page_title: product['page_title'] || '',
+                  preorder_message: self.get_availability(product) == 'preorder' ? product['availability'] : '',
+                  preorder_release_date: product['releaseDate'] && product['releaseDate'].to_datetime ? product['releaseDate'].to_datetime.strftime("%FT%T%:z") : nil,
+                  price: product['offers'] && product['offers'][0] ? product['offers'][0]['price'] : '0',
+                  search_keywords: self.get_search_keywords(additional_data.delete(:additional_search_terms), product),
+                  sku: product ? product['sku'] : default_sku,
+                  type: product['isDigital'] == true || is_digital ? 'digital' : 'physical',
```

Comment:
This looks like `is_digital` is always going to be true. I think we want to drop the `true ||`